### DISCUSSION TITLE
2.25.0

### DISCUSSION THREAD

**Discussion by Jolg42** [prisma/prisma#7683] [Q&A]
Today, we are excited to share the `2.25.0` stable release 🎉

🌟 **Help us spread the word about Prisma by starring the repo or [tweeting](https://twitter.com/intent/tweet?text=Check%20out%20the%20latest%20@prisma%20release%20v2.25.0%20%F0%9F%9A%80%0D%0A%0D%0Ahttps://github.com/prisma/prisma/releases/tag/2.25.0) about the release.** 🌟 

### Human-readable drift diagnostics for `prisma migrate dev`

Database schema drift occurs when your database schema is out of sync with your migration history, i.e. the database schema has _drifted away_ from the source of truth.

With this release, we improve the way how the drift is printed to the console when detected in the `prisma migrate dev` command. While this is the only command that uses this notation in today's release, we plan to use it in other places where it would be useful for debugging in the future.

Here is an example of how the drift is presented with the new format:

```
[*] Changed the `Color` enum
  [+] Added variant `TRANSPARENT`
  [-] Removed variant `RED`

[*] Changed the `Cat` table
  [-] Removed column `color`
  [+] Added column `vaccinated`

[*] Changed the `Dog` table
  [-] Dropped the primary key on columns (id)
  [-] Removed column `name`
  [+] Added column `weight`
  [*] Altered column `isGoodDog` (arity changed from Nullable to Required, default changed from `None` to `Some(Value(Boolean(true)))`)
  [+] Added unique index on columns (weight)
```

<details><summary>Expand to view the resulting Prisma schema</summary>

This file is generated to `prisma/schema.prisma`:

```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}
```

</details>

### Support for `.env` files in Prisma Client Go

You can now use a `.env` file with Prisma Client Go. This makes it easier to keep database credentials outside your Prisma schema and potentially work with multiple clients at the same time:

```
example/
├── .env
├── main.go
└── schema.prisma
```

Learn more about using the `.env` file in [our documentation](https://www.prisma.io/docs/concepts/more/environment-variables/#using-env-files).

### Dropping support for Node.js v10

Node.js v10 reached End of Life on April 30th 2021. Many of our dependencies have already dropped support for Node.js v10 so staying up-to-date requires us to drop Node.js v10, too.

We recommend upgrading to Node.js v14 or greater for long-term support. You can learn more about Node.js releases on [this page](https://nodejs.org/en/about/releases/).

### Prisma Client

- [[Errors] New common error from malformed connection string](https://github.com/prisma/prisma/issues/3405)
- [Failed to run future: no available capacity / QueueFull](https://github.com/prisma/prisma/issues/6683)
- [Clean up middleware implementation](https://github.com/prisma/prisma/issues/6968)
- [Possible Node API performance regression with concurrency](https://github.com/prisma/prisma/issues/7382)
- [Dropping Node v10 support (LTS reached EndOfLife)](https://github.com/prisma/prisma/issues/7394)
- [Performance regression with napi for high concurrency](https://github.com/prisma/prisma/issues/7404)
- [In 2.24.0 `createMany` errors with: `PANIC in query-engine/connectors/sql-query-connector/src/database/operations/write.rs:95:60called `Result::unwrap()` on an `Err` value: ScalarFieldNotFound { name: "updated_at", model: "FundMangerTeamMemberExperience" }`](https://github.com/prisma/prisma/issues/7416)
- [test: snapshot `params` in Middleware tests](https://github.com/prisma/prisma/issues/7437)
- [Running `prisma generate` doesn't install prisma client 2.24.x](https://github.com/prisma/prisma/issues/7579)
- [Middleware + $transactions do not play well together](https://github.com/prisma/prisma/issues/7584)
- [Environment variable not found](https://github.com/prisma/prisma/issues/7629)

### Prisma Migrate

- [Feature request - Migrate: When Drift Detected -> give more Details than just the rollback script](https://github.com/prisma/prisma/issues/5873)
- [Error: [introspection-engine\connectors\sql-introspection-connector\src\introspection_helpers.rs:188:64] called `Option::unwrap()` on a `None` value ](https://github.com/prisma/prisma/issues/6738)
- [Generate log file with Drift detected info for troubleshooting purpose](https://github.com/prisma/prisma/issues/6776)
- [`db push` / `migrate` is hiding the binary download process during `generate`, leading to confusion](https://github.com/prisma/prisma/issues/6926)
- [Shadow vs. main database check does not consider the port](https://github.com/prisma/prisma/issues/7216)
- [Re-Introspection removes `shadowDatabaseUrl` from schema](https://github.com/prisma/prisma/issues/7337)

### Prisma Studio

- [Prisma Studio not working with napi preview feature](https://github.com/prisma/studio/issues/662)
- [Crash in simple datasource only schema](https://github.com/prisma/studio/issues/701)
- [Prisma studio requires having prisma client installed when it shouldn't](https://github.com/prisma/studio/issues/703)

## Credits

Huge thanks to @82600417, @SuryaElavazhagan, @psavan1655 for helping!

## 📺 Join us for another "What's new in Prisma" livestream

Learn about the latest release and other news from the Prisma community by joining us for another ["What's new in Prisma"](https://www.youtube.com/watch?v=Fkj3Zaow5fQ&ab_channel=Prisma) livestream.

The stream takes place [on Youtube](https://www.youtube.com/watch?v=Fkj3Zaow5fQ&ab_channel=Prisma) on **Thursday, June 17** at **5pm Berlin | 8am San Francisco**.

## 🌎 Prisma Day is coming

Save the date for [Prisma Day 2021](https://www.prisma.io/day) and join us for two days of talks and workshops by the most exciting members of the Prisma community. 

- June 29th: Workshops
- June 30th: Talks

We look forward to seeing you there!

<hr /><em>This discussion was created from the release <a href='https://github.com/prisma/prisma/releases/tag/2.25.0'>2.25.0</a>.</em>

**Comment by willdoescode** [comment]
Swag

**Reply by psavan1655** [reply]
We'll be getting swags for the contribution? 😅

**Reply by Mehak-Mehta** [reply]
mega swag

**Comment by Linux249** [comment]
Hey, 
found "prisma client doesn't work with serverless-next.js" in the change log under "Fixes and improvements". I think this is misleading. The issue is "solved" cause there is a workaround described inside the issue to solve the problem -> ok. 
But reading your change log suggests to me that the problem would no longer exist when upgrading to 2.25.0 ... or is it me reading it wrong in general?

**Reply by janpio** [reply]
Good point, I removed that issue from the list. The list of issues in a mile stone is auto generated, so sometimes things we just close etc slip through and make it into this list.

We would of course prefer to _fully_ support this, but seems that is not easily possible in this case - so we should at least document it properly to make it more discoverable. If you are still having this problem and the workaround does not work (or you do not want to use it), feel free to open a new issue with a good description so we can take another look.

**Reply by Linux249** [reply]
Thanks for clarifying - it's a case that only appears when deploying a NextJs + Prisma app with serverless-nextjs to AWS so imho the serverless-nextjs team should mention the workaround in their docs too - will draw their attention to it.
Btw this ticket has the same topic https://github.com/prisma/prisma/issues/6032 ;)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
